DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/hard_drop_counters/test_hard_drop_counters.py

```diff
@@ -0,0 +1,396 @@
+
+    # 2.Configure ip address on ports and connect devices
+    addr_map = []
+    for p_indx in range(ports_num):
+        ip_addr = ip_network[1]
+        if ports_num == 1 or p_indx == 2:
+            addr_map.append((dut_ports[p_indx], tg_ports[p_indx], str(ip_addr + 1), str(ip_addr), ip_network.prefixlen,
+                             str(ip_network.broadcast_address), macs[dut_ports[p_indx]]))
+            addr_map.append((
+                dut_ports[p_indx + 1], tg_ports[p_indx + 1], '0.0.0.0',
+                '0.0.0.1', 24, '0.0.0.255', macs[dut_ports[p_indx + 1]]))
+        else:
+            addr_map.append((
+                dut_ports[p_indx], tg_ports[p_indx], str(ip_addr + 1),
+                str(ip_addr), ip_network.prefixlen, str(ip_network.broadcast_address), macs[dut_ports[p_indx]]))
+        ip_network = IPv4Network(f'{ip_network[0] + 256}/{ip_network.prefixlen}')
+
+    out = await IpAddress.add(input_data=[{dev_name: [
+        {'dev': port, 'prefix': f'{ip}/{plen}', 'broadcast': broadcast}
+        for port, _, _, ip, plen, broadcast, _ in addr_map]}])
+    assert not out[0][dev_name]['rc'], f'Failed to add IP addr to ports \n{out}'
+
+    dev_groups = tgen_utils_dev_groups_from_config(
+        [{'ixp': tg_port, 'ip': tg_ip, 'gw': gw, 'plen': plen}
+         for _, tg_port, tg_ip, gw, plen, _, _ in addr_map])
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, dut_ports, dev_groups)
+
+    # 3.Add Static Arp entry on the first and second ports connected to ixia
+    out = await IpNeighbor.add(input_data=[{dev_name: [
+        {'dev': port, 'address': nei_ip, 'lladdr': lladdr}
+        for port, _, nei_ip, _, _, _, lladdr in addr_map[:ports_num]
+    ]}])
+    assert not out[0][dev_name]['rc'], 'Failed to add static arp entries'
+
+    # 4.Set up streams matching all L3 invalid packets types on tx ports
+    streams = {}
+    for p_indx in range(ports_num):
+        src_dst_ports = [tg_ports[p_indx], tg_ports[p_indx ^ 1]]
+        port_streams = await get_illegal_drop_streams(dent_dev, dev_groups, src_dst_ports, dut_ports[p_indx],
+                                                      macs[dut_ports[p_indx]], addr_map[p_indx][2], size=frame_size)
+        streams.update(port_streams)
+    # Update ip_sip_is_zero_ when 1 port (tg_port[1] will be as tx_port)
+    if ports_num == 1:
+        swp_info = {}
+        await tgen_utils_get_swp_info(dent_dev, dut_ports[1], swp_info)
+        streams[f'ip_sip_is_zero_{dut_ports[0]}']['ip_source'] = dev_groups[tg_ports[1]][0]['name']
+        streams[f'ip_sip_is_zero_{dut_ports[0]}']['ip_destination'] = dev_groups[tg_ports[0]][0]['name']
+        streams[f'ip_sip_is_zero_{dut_ports[0]}']['dstMac'] = swp_info['mac']
+        streams[f'ip_sip_is_zero_{dut_ports[0]}']['srcMac'] = macs[dut_ports[1]]
+
+    # 5.Transmit traffic and verify drop counters updated according to the rate packets sent
+    for key in DROP_COUNTERS_MAP:
```

Comment:
Can we start all traffic items and verify all drop counters with one traffic send session?

Reply:
In theory, this should be possible (at least if cpu codes for traffic items do not overlap). @AndriyLozovyyPLV can you please check this?